src/ceph-volume/ceph_volume/devices/raw/list.py

```diff
@@ -36,18 +36,23 @@ def _get_bluestore_info(dev):
         logger.warning('skipping device {} because it is not reported in ceph-bluestore-tool output: {}'.format(dev, out))
         return None
     try:
-        if oj[dev]['description'] != 'main':
-            # ignore non-main devices, for now
-            logger.info('ignoring non-main device {}'.format(dev))
-            return None
-        whoami = oj[dev]['whoami']
-        return {
-            'type': 'bluestore',
-            'osd_id': int(whoami),
+        r = {
             'osd_uuid': oj[dev]['osd_uuid'],
-            'ceph_fsid': oj[dev]['ceph_fsid'],
-            'device': dev
         }
+        if oj[dev]['description'] == 'main':
```

Comment:
Can we list the db/wal by passing an argument to `list` so that we keep the current behavior? Not how various tooling will react if they get more devs and don't parse their type.

Reply:
@liewegas looks like this comment hasn't been addressed

Reply:
I'm not following what the "current behavior" is that you're worried about.  IIUC this should behave the same for current users.  The difference is that if db or wal devices are found they will be added into the appropriate dicts.  If only a db/wal device is found then you don't get all of the metadata because it is not all recorded in the db/wal device's label.

Reply:
There is a lot of code around the list. Today, every tool assumes that the list will return a list of OSDs and OSDs only. So if we get db/wal too they will be treated as OSDs and thus will break any implementation. So I was thinking that we could have a flag like `--all` if we want to print DB/WAL if any. Makes sense? Thanks!

Reply:
Do you mean we should exclude entries if there is a db/wal device but no primary device? 

Or do you mean we shouldn't include ``device_db`` and ``device_wal`` fields unless an extra flag is provided?

Reply:
The latter.

Reply:
Do you expect new keys in the json dictionary to break any existing users?  The ``device_db`` and ``device_wal`` keys were never there before.  Usually the use of dicts like this is intended to allow you to add new items without breaking existing users.

Reply:
It's essentially because they were not there before that they might cause breakage. Most of us today assume that the `c-v raw list` will list OSDs ID **only**. With this change, it will list OSD ID as well as metadata devices, which diverges from its original goal I believe. Toos currently listing and parsing entries will likely be confused and might find more OSDs than needed. That's why adding a flag like `--all` will not change the current listing behavior. If we need associate metadata devices to be displayed we need a flag for this.

We could have `--all` and/or even go more granular with a `--block-db` and `--block-wal` to individually find db or wal devices.

Reply:
I'm confused by "will ost OSDs ID only". 

Before:
```
{
    "2414f53f-d2cb-47ad-93c5-6db12e5cbd18": {
        "ceph_fsid": "1f876d56-f4f1-4ebd-b4b5-4f9a638a7fed",
        "device": "/dev/mapper/ceph--9950fec4--f3db--41b8--bb8e--4319956c50b6-osd--block--2414f53f--d2cb--47ad--93c5--6db12e5cbd18",
        "osd_id": 33,
        "osd_uuid": "2414f53f-d2cb-47ad-93c5-6db12e5cbd18",
        "type": "bluestore"
    },
    "30623e88-1e3b-4419-9e98-6947ed5a5ebc": {
        "ceph_fsid": "1f876d56-f4f1-4ebd-b4b5-4f9a638a7fed",
        "device": "/dev/mapper/ceph--d702c1db--9311--4d1b--ac1b--99779aa2b5f4-osd--block--30623e88--1e3b--4419--9e98--6947ed5a5ebc",
        "osd_id": 36,
        "osd_uuid": "30623e88-1e3b-4419-9e98-6947ed5a5ebc",
        "type": "bluestore"
    },
...
```
With this PR:
```
{
    "2414f53f-d2cb-47ad-93c5-6db12e5cbd18": {
        "ceph_fsid": "1f876d56-f4f1-4ebd-b4b5-4f9a638a7fed",
        "device": "/dev/mapper/ceph--9950fec4--f3db--41b8--bb8e--4319956c50b6-osd--block--2414f53f--d2cb--47ad--93c5--6db12e5cbd18",
        "device_db": "/dev/mapper/ceph--65408994--5401--4293--a75e--59d6d7f8440c-osd--db--fd8b973f--b284--4546--859a--6e9dde14cba0",
        "osd_id": 33,
        "osd_uuid": "2414f53f-d2cb-47ad-93c5-6db12e5cbd18",
        "type": "bluestore",
        "whoami": "33"
    },
    "30623e88-1e3b-4419-9e98-6947ed5a5ebc": {
        "ceph_fsid": "1f876d56-f4f1-4ebd-b4b5-4f9a638a7fed",
        "device": "/dev/mapper/ceph--d702c1db--9311--4d1b--ac1b--99779aa2b5f4-osd--block--30623e88--1e3b--4419--9e98--6947ed5a5ebc",
        "device_db": "/dev/mapper/ceph--0c7ee120--ce01--40c7--874c--5af448ecd806-osd--db--f22bb363--3cd5--4f6e--8791--9645db6cbbc2",
        "osd_id": 36,
        "osd_uuid": "30623e88-1e3b-4419-9e98-6947ed5a5ebc",
        "type": "bluestore",
        "whoami": "36"
    },
...
```